### PR TITLE
Profile audience and scope

### PR DESCRIPTION
**Pull request by ghost**
Revised wording to clarify scope to include processes to registration/pre-authorization backend service, and for runtime acquisition of access token.  Explicitly stated that profile's applicability is not restricted to  retrieval of bulk data.  Clarified conditions under which profile applies.

### CODE REVIEW

File: authorization.md [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
 
 This profile is intended to be used by developers of back-end services that
-need to access FHIR resources by requesting access tokens from OAuth 2.0
-compliant authorization servers. This profile assumes that a backend service
-has been authorized up-front, at registration time, and describes the runtime
-process by which the service acquires an access token that can be used to
-communicate with a FHIR Resoure Server.
+autonomously (or semi-autonomously) need to access FHIR resources from an EHR 
+that has pre-authorized a defined scope of access.  Specifically, this profile 
+describes the registration process by which a backend service is pre-authorized,
```
Comment (michelemottini) [inline authorization.md]: It does not really describe the registration process, does it?

Comment (jmandel) [reply authorization.md]: It describes functional requirements for the process; would you prefer different wording here?

Comment (michelemottini) [reply authorization.md]: I prefer the old wording `This profile assumes that a backend service
has been authorized up-front`

Comment (jmandel) [reply authorization.md]: OK. @bakerdb perhaps we can use a hybrid like:

```suggestion
describes the registration-time metadata required for a backend service to be pre-authorized,
```

Comment (ghost) [reply authorization.md]: I don't think we provide much detail regarding the registration process at all.   We just require that the backend client has pre-registered and that the EHR has authorized a scope of access.

**Comment by jmandel** [comment]
Wait, did you mean to close the entire PR?

**Comment by ghost** [comment]
No.  Trying to recall how GitHub works  when a lot of discussion is involved.  I may need a refresher - or at least patience. 😕

- Dixie


Dixie Baker
Office: 310-791-9671
Cell: 310-279-2579


On Oct 29, 2018, at 6:58 PM, Josh Mandel <notifications@github.com<mailto:notifications@github.com>> wrote:


Wait, did you mean to close the entire PR?

**Comment by jmandel** [comment]
I think we can help with both :-) to start, I'll hit the "reopen" button.

File: authorization.md (hunk 1 above)
Comment (ghost) [reply authorization.md]: I like your suggested wording here, Josh.

Comment (jmandel) [reply authorization.md]: Cool  -- I think you can click to accept the proposed change (this is a new feature in GitHub; if you don't see it, just let me know and I'll apply directly).

Comment (ghost) [reply authorization.md]: I see the "Apply Suggestion" button.  But when I click it, it says "Outdated Suggestions cannot be applied."  Guess I took too long.

Comment (jmandel) [reply authorization.md]: Oh, ok -- you successfully accepted the suggestion yesterday, and it's now reflected in this PR. So... good :-)

**Comment by isaacvetter** [comment]
Hey Guys,

It looks like this PR is introducing a couple of sentences that really strongly suggest that backend services and bulk data only work with an EHR as the server; yet, our existing use-case descriptions clearly support other scenarios. 

Is it reasonable to not suggest that an EHR is always the bulk data server?

Isaac

Examples:
```
his profile is intended to be used by developers of back-end services that autonomously (or semi-autonomously) need to access FHIR resources from an EHR 
```

```
 The target EHR can register the backend service and pre-authorize access to a 
```


From this [diff](https://github.com/smart-on-fhir/fhir-bulk-data-docs/pull/86/files)

**Comment by michelemottini** [comment]
It's what also the SMART specs use. It is not exact, but not sure what would be a good alternative

**Comment by jmandel** [comment]
In the intro scope paragraph @isaacvetter I'd propose adding something like:

> (This specification uses the term "EHR" to refer to any data system that hosts FHIR data in the role of
a bulk data server; it could include clinical data systems, financial data systems,
and other health information systems.)

Does this work for you?

**Comment by ghost** [comment]
@jmandel @isaacvetter  The problem I see with the proposed clarification is that we have defined the scope of this specification as specifically NOT being limited to bulk-data accesses, but rather in terms of the client having been pre-authorized to access the requested FHIR data.  I recommend replacing the term "EHR" with "FHIR server" throughout (or FHIR resource server, FHIR authorization server where that more specificity is needed).  Alternatively, we could use Josh's suggested wording and just drop "in the role of a bulk data server."  My preference would be the former.

**Comment by jmandel** [comment]
Indeed --- if nothing else I should have written "in the role of a FHIR server". I would also be happy replacing the terms throughout though, as @bakerdb suggests.

**Comment by jmandel** [comment]
Quick follow-up on this one @bakerdb  -- do you still want to apply the change from "EHR" to "FHIR Server" in this PR? (I'm happy to take care of the mechanics, or you can; just let me know.)

**Comment by ghost** [comment]
I'm in the process of changing both "backend server" to "client" (another conversation) and "EHR" to "FHIR server."  I'll make these changes in both merged content and open pull requests, as appropriate, to minimize confusion.

**Comment by ghost** [comment]
I have completed going through both specifications (merged content and open PRs) to make sure the "client" and "FHIR server" language is used consistently.

**Pull request merged**